src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -26,6 +26,7 @@ public unsafe static uint[] Square(uint[] value)
 
         // Mutable for unit testing...
         private static int SquareThreshold = 32;
+        private static int AllocationThreshold = 4096;
```

Comment:
I have to say 4096 seems large too me. Once the number is large enough, the cost of allocating it on the heap is probably dwarfed by the cost to process the number. Unfortunately, I am not sure where the cut off point should be. Maybe @Maoni0, or @pdussud could suggest something.